### PR TITLE
Changed make to support more gpu archs, multiple toolkits, reduce lib size.

### PR DESCRIPTION
**Pull request by DickJC123**

### CODE REVIEW

File: make/config.mk [hunk 1]
```diff
@@ -54,12 +54,29 @@ USE_CUDA_PATH = NONE
...
+# of the cuda toolkit, plus the assembly (PTX) for the most recent such architecture.
+# If these kernels are then run on a newer-architecture GPU, then the binary will
+# be JIT-compiled by the updated driver from the included PTX.
+#
+# If CUDA is not present on the system, CUDA_ARCH will be quietly set to "".
+KNOWN_CUDA_ARCHS := 30 35 50 52 60 61
+# Run nvcc on a zero-length file to check architecture-level support.
+# Build nvcc arg list for supported levels to include SASS in the fat-binary.
+CUDA_ARCH := $(foreach arch,$(KNOWN_CUDA_ARCHS), \
+                  $(shell nvcc -arch=sm_$(arch) -E --x cu /dev/null >/dev/null 2>&1 && \
```
Comment (piiswrong) [inline make/config.mk]: what if there is no nvcc / fatbinary?

Comment (mli) [inline make/config.mk]: leave `CUDA_ARCH` as empty in `config.mk`. In makefile, if this variable is empty, then assign it with values by using `$(NVCC)`

Comment (DickJC123) [inline make/config.mk]: My recent commit incorporates these comments/suggestions.  CUDA_ARCH is now set only if USE_CUDA == 1.  I also eliminated direct use of the fatbinary program, accomplishing the same test of compression support through $(NVCC) use directly (and now in the Makefile, not in any of the make/*.mk files).  Users can alter the arch support by adding lines to the end of config.mk, or through environment variables, as in:

export CUDA_ARCH="-gencode arch=compute_61,code=sm_61"

**Comment by mli** [comment]
Is the goal only building the arch for the GPU installed to accelerate the building? I checked the CI build, it still has all arches enabled `-gencode arch=compute_30,code=sm_30 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=[sm_52,compute_52] `

**Comment by mli** [comment]
Figured out the answer in the comment. This PR LGTM

**Pull request merged**